File: sum_of_primes/sum_of_primes.rb
# SUM OF PRIMES
# CHALLENGE DESCRIPTION:

# Write a program which determines the sum of the first 1000 prime numbers.

# INPUT SAMPLE:

# There is no input for this program.

# OUTPUT SAMPLE:

# Print to stdout the sum of the first 1000 prime numbers.

# 3682913

def sum_of_primes
  primes, n = [], 2
  until primes.size == 1000
    divisors = []
    1.upto(n) do |i|
      divisors << i if n % i == 0
      break if divisors.size > 2
    end
    primes << n if divisors.size == 2
    n += 1
  end
  primes.reduce :+
end

puts sum_of_primes

File: sum_of_primes/sum_of_primes.go
package main

import(
  "fmt"
)

func main() {
  var primes []int
  n := 2
  for len(primes) < 1000 {
    is_prime := true

    for m := 2; m <= n / 2; m++ {
      if n % m == 0 {
        is_prime = false
      }
    }

    if is_prime {
      primes = append(primes, n)
    }

    n += 1
  }

  sum := 0
  for i := 0; i < len(primes); i++ {
    sum += primes[i]
  }

  // fmt.Printf("primes: %v\n", primes)
  // fmt.Printf("length: %v\n", len(primes))
  // fmt.Printf("sum: %v\n", sum)
  fmt.Printf("%v", sum)
}